dev/tools/pubspec.yaml

```diff
@@ -29,12 +29,43 @@ dependencies:
   typed_data: 1.1.6 # THIS LINE IS AUTOGENERATED - TO UPDATE USE "flutter update-packages --force-upgrade"
 
 dev_dependencies:
+  test: 1.6.3
```

Comment:
I'm also a little confused about how `pubspec` and `.packages` works -- To use some of the packages when running my tests, I had to manually add `test:file:///{user}/.pub-cache/hosted/pub.dartlang.org/test-1.6.3/lib/` to the .packages file in `dev/tools/localizations`. Is this because the test file is not being referenced by some file in `dev/tools`?

Reply:
I'm guessing I'll also have to hook these tests up to the CI somehow

Reply:
when you invoke `dart` it starts looking for a .packages file in current working directory - searches up a directory level until it finds it. Once it is located, the .packages file is used to resolve the package uris using the file paths relative to the current location of the .packages file.

The pubpsec file generates a .packages file in the same directory it is in. This is a destructive process, so If you've manually edited a .packages file it will be removed.

For hooking it up to CI, try looking in `dev/bots/test.dart`